src/polytopes/chain.jl

```diff
@@ -4,59 +4,31 @@
 
 """
     Chain(p1, p2, ..., pn)
+    Chain(points::AbstractVector)
 
-A polygonal chain from a sequence of points `p1`, `p2`, ..., `pn`.
+A polygonal chain from a sequence of points `p1`, `p2`, ..., `pn`. If `points`
+is of type `CircularVector`, the chain is considered closed.
 See https://en.wikipedia.org/wiki/Polygonal_chain.
-"""
-struct Chain{Dim,T,V<:AbstractVector{Point{Dim,T}}} <: Polytope{1,Dim,T}
-  vertices::V
-end
-
-# specialize Polytope outer constructor to use standard vector
-Chain(vertices::Vararg{P}) where {P<:Point} = Chain(collect(vertices))
-
-Chain(vertices::CircularVector) = Chain([collect(vertices); first(vertices)])
-
-"""
-    npoints(chain)
-
-Return the total number of points used to represent the
-chain, no matter if it is closed or open.
-
-See also [`nvertices`](@ref).
 
 ### Notes
 
-This function is provided for IO purposes. Most algorithms
-should be written in terms of `nvertices` and `vertices`
-as they are consistent with each other.
+- `Ring` is a type alias for closed chains.
 """
-npoints(c::Chain) = length(c.vertices)
-
-"""
-    nvertices(chain)
+struct Chain{Dim,T,V<:AbstractVector{Point{Dim,T}}} <: Polytope{1,Dim,T}
+  vertices::V
+end
 
-Return the number of vertices of the `chain`. In the case
-that the chain is closed, the number of vertices is one
-less the total number of points used to represent the chain.
 """
-nvertices(c::Chain) = npoints(c) - isclosed(c)
+    Ring(p1, p2, ..., pn)
+    Ring(points::AbstractVector)
 
+A closed polygonal chain from a sequence of points `p1`, `p2`, ..., `pn`. This
+is a type alias for a `Chain` with a `CircularVector`.
 """
-    vertices(chain)
+const Ring = Chain{Dim,T,V} where {Dim,T,V<:CircularVector{Point{Dim,T}}}
 
-Return the vertices of the `chain`. In the case that the
-chain is closed, the returned vector is circular and can
-be indexed with arbitrarily negative or positive indices.
-"""
-function vertices(c::Chain)
-  if isclosed(c)
-    vs = @view c.vertices[begin:end-1]
-    CircularVector(vs)
-  else
-    c.vertices
-  end
-end
+(::Type{Ring})(vertices::AbstractVector) = Chain(CircularVector(vertices))
+(::Type{Ring})(vertices::CircularVector) = Chain(vertices)
```

Comment:
Isn't the syntax Ring(vertices::AbstractVector) supported? Just double checking if we can remove the ::Type wrapper.

Reply:
Done in newer version.